awx/ui_next/src/screens/Template/JobTemplateDetail/JobTemplateDetail.jsx

```diff
@@ -386,22 +388,22 @@ function JobTemplateDetail({ i18n, template }) {
             <DeleteButton
               name={name}
               modalTitle={i18n._(t`Delete Job Template`)}
-              onConfirm={handleDelete}
+              onConfirm={deleteJobTemplate}
```

Comment:
Perhaps instead of hiding the whole UI while the delete is loading (line 161), you could just disable this button temporarily. That way, if an error occurs, the screen doesn't disappear then reappear.

`<DeleteButton ... isDisabled={isLoading}>`

You may also need to update the DeleteButton to pass this prop through to the PF Button.